gatsby-ssr.tsx

```diff
@@ -71,24 +73,35 @@ export const onPreRenderHTML: GatsbySSR['onPreRenderHTML'] = ({
     return
   }
 
-  const transformedHeadComponents = getHeadComponents().map((node) => {
-    if (isStyleComponent(node)) {
-      const globalStyleHref = node.props?.['data-href'] ?? node.props?.href
+  const transformedHeadComponents = getHeadComponents()
+    .sort((node) => {
+      if (isStyleComponent(node)) {
+        const styleHref = node.props?.['data-href'] ?? node.props?.href ?? ''
+        const isGlobalStyle = /^\/styles.[a-zA-Z0-9]*.css$/.test(styleHref) // global style regex
```

Comment:
I was thinking about that sorting problem when importing CSS files inside its components. Can we always sort all `css` files here to avoid this problem and keep the imports at the components?

Reply:
This sort here and the problem we had with CSS conflicts are different things. Here we're sorting the `<link rel="stylesheet">` tags that go inside the `<head>`. to make sure the global CSS file comes first, so the rest can override when needed. Here, at his level, we cannot sort the CSS imports inside the components.
:\

Reply:
This sort here is just to keep the global CSS always like the first. I think dealing with the component styles here is not the best approach. We can discuss more as well :)